src/panels/config/automation/ha-automation-editor.ts

```diff
@@ -421,7 +423,9 @@ export class HaAutomationEditor extends KeyboardShortcutMixin(LitElement) {
         </div>
         <ha-fab
           slot="fab"
-          class=${classMap({ dirty: !this._readOnly && this._dirty })}
+          class=${classMap({
+            dirty: !this._readOnly && (this._dirty || !!this._yamlErrors),
```

Comment:
I think it would be cleaner to do `this._dirty = true` in `_yamlChanged`

Reply:
good idea